libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -238,6 +239,33 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         navigationOptions.eHorizonOptions.minTimeDeltaBetweenUpdates
     )
 
+    // native Router Interface
+    private val nativeRouter: RouterInterface by lazy {
+        NavigatorLoader.createNativeRouterInterface(
+            navigationOptions.deviceProfile,
+            navigatorConfig,
+            createTilesConfig(
+                isFallback = false,
+                tilesVersion = navigationOptions.routingTilesOptions.tilesVersion
+            ),
+            historyRecorder.fileDirectory(),
+        )
+    }
+
+    // Router provided via @Modules, might be outer
+    private val moduleRouter: Router by lazy {
+        MapboxModuleProvider.createModule(MapboxModuleType.NavigationRouter) {
+            paramsProvider(
+                ModuleParams.NavigationRoute(
+                    accessToken
+                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ROUTER),
+                    nativeRouter,
+                    threadController
+                )
+            )
+        }
+    }
```

Comment:
```suggestion
    private val moduleRouter: NavigationRouter by lazy {
        val result = MapboxModuleProvider.createModule<Router>(MapboxModuleType.NavigationRouter) {
            paramsProvider(
                ModuleParams.NavigationRoute(
                    accessToken
                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ROUTER),
                    nativeRouter,
                    threadController
                )
            )
        }
        if (result is NavigationRouter) {
            result
        } else {
            LegacyRouterAdapter(result)
        }
    }
```

This should allow you to remove the custom mappings that you have in the `RouterInterfaceDelegate`. Just use the `DirectionsResponse` from the `routes` returned in the `NavigationRouterCallback`.